pkgs/top-level/ocaml-packages.nix

```diff
@@ -519,6 +519,8 @@ let
 
     pgocaml = callPackage ../development/ocaml-modules/pgocaml {};
 
+    ocaml_sat_solvers = callPackage ../development/ocaml-modules/sat-solvers { };
```

Comment:
This attribute and the directory containing the package should have the same name: `ocaml-sat-solvers` (the one you have used as value for “pname”, which is also the name used by findlib/ocamlfind…).